Framework/HistogramData/test/RebinTest.h

```diff
@@ -0,0 +1,446 @@
+#ifndef MANTID_HISTOGRAMDATA_REBINTEST_H_
+#define MANTID_HISTOGRAMDATA_REBINTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidHistogramData/Histogram.h"
+#include "MantidHistogramData/LinearGenerator.h"
+#include "MantidHistogramData/Rebin.h"
+
+#include <algorithm>
+#include <random>
+
+using namespace Mantid::HistogramData;
+
+class RebinTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static RebinTest *createSuite() { return new RebinTest(); }
+  static void destroySuite(RebinTest *suite) { delete suite; }
+
+  void testExecrebin() {
+    TS_ASSERT_THROWS_NOTHING(
+        rebin(getCountsHistogram(), BinEdges(10, LinearGenerator(0, 0.5))));
+  }
+
+  void testExecRebinFrequency() {
+    TS_ASSERT_THROWS_NOTHING(
+        rebin(getFrequencyHistogram(), BinEdges(10, LinearGenerator(0, 0.5))));
+  }
+
+  void testRebinNoYModeDefined() {
+    BinEdges edges(5, LinearGenerator(0, 2));
+    // X Mode Points
+    TS_ASSERT_THROWS(
+        rebin(Histogram(Histogram::XMode::Points, Histogram::YMode::Counts),
+              edges),
+        std::runtime_error);
```

Comment:
Strictly speaking you don't know if this throws because of the XMode (without looking into the code). Could be anything, e.g., the `nullptr` you have. Can you pass a valid `Histogram` instead?